Applications/shapeworks/Commands.cpp

```diff
@@ -1555,91 +1654,948 @@ bool Coverage::execute(const optparse::Values &options, SharedCommandData &share
+  parser.add_option("--originz", "-z").action("store").type("double").set_default(0.0).help("Origin about which reflection occurs in z-direction [default: %default].");
+
+  Command::buildParser();
+}
+
+bool ReflectMesh::execute(const optparse::Values &options, SharedCommandData &sharedData)
+{
+  if (!sharedData.validMesh())
+  {
+    std::cerr << "No mesh to operate on\n";
+    return false;
   }
-  else {
-    OptimizeParameterFile param;
-    param.load_parameter_file(project_file.c_str(), &app);
-    return app.Run();
+
+  std::string axis_str(static_cast<std::string>(options.get("axis")));
+  Axis axis(toAxis(axis_str));
+
+  double originX = static_cast<double>(options.get("originx"));
+  double originY = static_cast<double>(options.get("originy"));
+  double originZ = static_cast<double>(options.get("originz"));
+
+  if (!axis_is_valid(axis))
+  {
+    std::cerr << "Must specify a valid axis (X, Y, or Z)\n";
+    return false;
+  }
+  else
+  {
+    sharedData.mesh->reflect(axis, makeVector({originX, originY, originZ}));
+    return sharedData.validMesh();
+  }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Transform
+///////////////////////////////////////////////////////////////////////////////
+void Transform::buildParser()
+{
+  const std::string prog = "transform-mesh";
+  const std::string desc = "transform mesh to target mesh using specified method (default: iterative closest point (ICP) 3D rigid registration)";
+  parser.prog(prog).description(desc);
+
+  parser.add_option("--target").action("store").type("string").set_default("").help("Filename of target mesh.");
+  std::list<std::string> aligns{"rigid", "similarity", "affine"};
+  parser.add_option("--type").action("store").type("choice").choices(aligns.begin(), aligns.end()).set_default("similarity").help("Alignment type to use [default: %default].");
+  std::list<std::string> methods{"icp"};
+  parser.add_option("--method").action("store").type("choice").choices(methods.begin(), methods.end()).set_default("icp").help("Method used to compute transform [default: %default].");
+  parser.add_option("--iterations").action("store").type("unsigned").set_default(10).help("Number of iterations run [default: %default].");
+
+  Command::buildParser();
+}
+
+bool Transform::execute(const optparse::Values &options, SharedCommandData &sharedData)
+{
+  if (!sharedData.validMesh())
+  {
+    std::cerr << "No mesh to operate on\n";
+    return false;
+  }
+
+  std::string targetMesh = static_cast<std::string>(options.get("target"));
+  unsigned iterations = static_cast<unsigned>(options.get("iterations"));
+  std::string alignopt(options.get("type"));
+  Mesh::AlignmentType align;
+  if (alignopt == "rigid")
+    align = Mesh::Rigid;
+  else if (alignopt == "similarity")
+    align = Mesh::Similarity;
+  else
```

Comment:
this should be `else if (alignopt == "affine")`, and abort if its none of the three. Right now if the user does "simlarity" or some typo it'll silently use affine.

Reply:
Thanks!

Reply:
This is now fixed in the mesh_party branch.